python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py

```diff
@@ -932,8 +932,20 @@ def evaluate(self, data, batch_size=32, multioutput="raw_values", quantize=False
                 invalidInputError(False,
                                   "You must call fit or restore first before calling evaluate!")
             if isinstance(data, DataLoader):
-                input_data = data
-                target = np.concatenate(tuple(val[1] for val in data), axis=0)
+                from torch.utils.data.sampler import RandomSampler
+                if isinstance(data.sampler, RandomSampler):
+                    # If dataloader is shuffled, convert input_data to numpy()
+                    # Avoid to iterate shuffled dataloader two times
+                    input_x = []
+                    input_y = []
+                    for val in data:
+                        input_x.append(val[0].numpy())
+                        input_y.append(val[1].numpy())
+                    input_data = np.concatenate(input_x, axis=0)
+                    target = np.concatenate(input_y, axis=0)
+                else:
+                    input_data = data
+                    target = np.concatenate(tuple(val[1] for val in data), axis=0)
```

Comment:
We should implement a way to "update" the metric like the implementation in https://torchmetrics.readthedocs.io/en/stable/classification/accuracy.html#functional-interface to continually calculate the metric rather than have a large numpy ndarray in later PR. Maybe we need an issue for this.

Reply:
yeah, I think we can do it later